habitat/tasks/nav/object_nav_task.py

```diff
@@ -23,6 +23,15 @@
 )
 
 
+@attr.s(auto_attribs=True, kw_only=True)
+class ObjectGoalNavEpisode(NavigationEpisode):
+    r"""ObjectGoal Navigation Episode
+
+    :param goals_key: Key to retrieve goals with
+    """
+    goals_key: str = None
```

Comment:
We need this key only once. I would use function/labda instead and avoid creating it.

Reply:
You need it all the time.  We need to save out the `goals_key` to disk.  The de-duplicated dataset should be saved to disk for the best performance benefits (for reference, running de-dup takes 30 minutes, so it really should be done just once).